parlai/core/context.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env python3
+
+# Copyright (c) 2017-present, Facebook, Inc.
+# All rights reserved.
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree. An additional grant
+# of patent rights can be found in the PATENTS file in the same directory.
+
+
+class BaseContext():
+    """Any XContext should implement the following set of functions. It can
+    then choose to implement any utility functions that it believes a model
+    especially tuned to use that context type will find useful.
+    """
+    def __init__(self, args):
+        """This function is used to directly create a context object"""
+        raise NotImplementedError
+
+    @staticmethod
+    def from_data(input_data):
+        """This function should take an input data object and produce the
+        context representation for this data. It is expected to be used by
+        tasks that are constructing context, and should be the inverse of
+        to_data_form
+        """
+        raise NotImplementedError
+
+    def to_data_form(self):
```

Comment:
What do you think about serialize/deserialize?

Reply:
I'll change to those - they tend to be more standard and definitionally correct.